packages/hpe-icons/src/js/grommet/utils.ts

```diff
@@ -0,0 +1,154 @@
+import { useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+
+declare module 'styled-components' {
+  export interface DefaultTheme {
+    icon?: {
+      disableScaleDown?: boolean;
+      size?: Record<string, string>;
+    };
+    text?: Record<string, { height: string }>;
+  }
+}
+
+export function isObject(item: unknown): item is Record<string, unknown> {
+  return !!item && typeof item === 'object' && !Array.isArray(item);
+}
+
+type ThemeObject = {
+  [key: string]: unknown;
+};
+
+export const deepMerge = (
+  target: ThemeObject,
+  ...sources: Partial<ThemeObject>[]
+): ThemeObject => {
+  if (!sources.length) {
+    return target;
+  }
+  // making sure to not change target (immutable)
+  const output = { ...target };
+  sources.forEach(source => {
+    if (isObject(source)) {
+      Object.keys(source).forEach(key => {
+        if (isObject(source[key])) {
+          if (!output[key]) {
+            output[key] = { ...source[key] };
+          } else {
+            output[key] = deepMerge(
+              output[key] as ThemeObject,
+              source[key] as ThemeObject,
+            );
+          }
+        } else {
+          output[key] = source[key];
+        }
+      });
+    }
+  });
+  return output;
+};
+
+export const parseMetricToNum = (string = '') =>
+  parseFloat(string.match(/\d+(\.\d+)?/)?.[0] || '0');
+
+// scaleProps sets path properties to prevent scaling the stroke
+// when the theme doesn't want it for small sizes.
+interface ScaleProps {
+  size: string;
+}
+
+interface ScaleResult {
+  vectorEffect?: string;
+}
+
+export function useScaleProps(props: ScaleProps): ScaleResult {
```

Comment:
Wondering if we can remove this function? Doesn't look like we are using it

Reply:
Can do. Would be good to eliminate if we can.